lib/very_good_infinite_list.dart

```diff
@@ -193,7 +212,8 @@ class _InfiniteListState<T> extends State<InfiniteList<T>> {
     super.initState();
     _controller = _ListController<T>(widget.itemLoader);
     _debouncer = _Debouncer(delay: widget.debounceDuration);
-    _scrollController.addListener(_onScroll);
+    _scrollController = widget._scrollController ?? ScrollController()
```

Comment:
If a `ScrollController` is provided, I think it should be up to the consumer to handle disposing the controller. The `InfiniteList` state should only handle disposing the controller if it was created internally.

Also, I think we need to handle scenarios where the injected `ScrollController` was updated in `didUpdateWidget`.

Let me know what you think 👍